test/gateway/eip20_gateway/confirm_revert_redeem_intent.js

```diff
@@ -0,0 +1,349 @@
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+const Gateway = artifacts.require("./TestEIP20Gateway.sol");
+const MockToken = artifacts.require("MockToken");
+
+const BN = require('bn.js');
+const EventDecoder = require('../../test_lib/event_decoder.js');
+const messageBus = require('../../test_lib/message_bus.js');
+const Utils = require('../../../test/test_lib/utils');
+const cogatewayUtils = require('../../../test/gateway/eip20_cogateway/helpers/co_gateway_utils');
+const StubData = require('../../data/redeem_revoked_1.json');
+
+const NullAddress = Utils.NULL_ADDRESS;
+
+async function setStorageRoot(gateway) {
+  let blockHeight = new BN(StubData.co_gateway.revert_redeem.proof_data.block_number, 16);
+  let storageRoot = StubData.co_gateway.revert_redeem.proof_data.storageHash;
+  await gateway.setStorageRoot(blockHeight, storageRoot);
+  return blockHeight;
+}
+
+contract('EIP20Gateway.confirmRevertRedeemIntent()', function (accounts) {
+
+  let gateway, mockToken, baseToken, unstakeRequest, unstakeMessage,
+    stakeVaultAddress;
+  let bountyAmount = new BN(StubData.gateway.constructor.bountyAmount, 16);
+  let MessageStatusEnum = messageBus.MessageStatusEnum;
+  let redeemRequest = StubData.co_gateway.redeem.params;
+
+  beforeEach(async function () {
+    redeemRequest = StubData.co_gateway.redeem.params;
+    mockToken = await MockToken.new({from: accounts[0]});
+    baseToken = await MockToken.new({from: accounts[0]});
+
+    let organizationAddress = accounts[3];
+    let coreAddress = accounts[5];
+    gateway = await Gateway.new(
+      mockToken.address,
+      baseToken.address,
+      coreAddress,
+      bountyAmount,
+      organizationAddress,
+      NullAddress //burner address,
+    );
+
+    unstakeRequest = {
+      beneficiary: redeemRequest.beneficiary,
+      amount: new BN(redeemRequest.amount, 16),
+    };
+
+    let redeemIntentHash = cogatewayUtils.hashRedeemIntent(
+      unstakeRequest.amount,
+      unstakeRequest.beneficiary,
+      StubData.contracts.coGateway
+    );
+
+    unstakeMessage = {
+      intentHash: redeemIntentHash,
+      nonce: new BN(redeemRequest.nonce, 16),
+      gasPrice: new BN(redeemRequest.gasPrice, 16),
+      gasLimit: new BN(redeemRequest.gasLimit, 16),
+      unstakeAccount: redeemRequest.redeemer,
+      hashLock: redeemRequest.hashLock,
+      unlockSecret: redeemRequest.unlockSecret,
+    };
+    stakeVaultAddress = await setup(unstakeMessage, gateway, unstakeRequest, stakeVaultAddress, mockToken, accounts);
+
+  });
+
+  it('should confirm revert redeem intent', async function () {
+
+    await gateway.setInboxStatus(unstakeMessage.messageHash, MessageStatusEnum.Declared);
+
+    let blockHeight = await setStorageRoot(gateway);
```

Comment:
This can go in to before each. Its used in 8 tests. 
`let blockHeight = await setStorageRoot(gateway);`

You can override the value in one of the test cases.